packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -366,6 +407,8 @@ export class ModularChangeFamily
 			composedNodeChange.nodeExistsConstraint = nodeExistsConstraint;
 		}
 
+		const key = change2 ?? change1 ?? fail("Should not compose two undefined changesets");
```

Comment:
I'm guessing this follows the same reasoning as documented above. Perhaps it would be best to put logic in function that it's not duplicated and the doc can be encapsulated in the function, so that one doesn't exist without the other.